src/MessageQuery.php

```diff
@@ -189,7 +191,8 @@ public function findOrFail(int $id, ImapFetchIdentifier $identifier = ImapFetchI
     public function find(int $id, ImapFetchIdentifier $identifier = ImapFetchIdentifier::Uid): ?MessageInterface
     {
         /** @var UntaggedResponse $response */
-        if (! $response = $this->id($id, $identifier)->first()) {
+        $response = $this->id($id, $identifier)->first();
```

Comment:
And that's your approach. I changed it to have $response directly at the DocTag. It's easier to understand this way... and PHPStan understands it this way